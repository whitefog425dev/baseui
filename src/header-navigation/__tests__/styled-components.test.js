/*
Copyright (c) 2018 Uber Technologies, Inc.

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.
*/
// @flow
import React from 'react';
import {shallow, mount} from 'enzyme';

import {
  StyledRoot,
  StyledNavigationList,
  StyledNavigationItem,
  ALIGN,
} from '../index.js';

describe('Header Navigation styled components', () => {
  describe('StyledRoot', function() {
    test('Root', () => {
      const component = shallow(
        <StyledRoot>
          <div />
        </StyledRoot>,
      );
      expect(component.instance().getStyles()).toMatchSnapshot(
        'StyledRoot has correct styles',
      );
    });
  });
  describe('NavigationList', function() {
    test('NavigationList', () => {
      const component = mount(
        <StyledNavigationList>
          <div />
        </StyledNavigationList>,
      );
      Object.values(ALIGN).forEach(align => {
        component.setProps({
          align,
        });
        expect(component.instance().getStyles()).toMatchSnapshot(
          // $FlowFixMe
          `NavigationList has correct styles when set to align ${align}`,
        );
      });
    });
  });
  describe('NavigationItem', function() {
    test('NavigationItem', () => {
      const component = shallow(
        <StyledNavigationItem>
          <div />
        </StyledNavigationItem>,
      );
      expect(component.instance().getStyles()).toMatchSnapshot(
        'NavigationItem has correct styles',
      );
    });
  });
});
