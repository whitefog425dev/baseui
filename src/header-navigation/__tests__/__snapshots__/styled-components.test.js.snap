// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Navigation styled components NavigationItem NavigationItem: NavigationItem has correct styles 1`] = `
Object {
  "alignSelf": "center",
  "paddingLeft": "$theme.sizing.scale800",
}
`;

exports[`Header Navigation styled components NavigationList NavigationList: NavigationList has correct styles when set to align center 1`] = `
Array [
  "align",
]
`;

exports[`Header Navigation styled components NavigationList NavigationList: NavigationList has correct styles when set to align flex-end 1`] = `
Array [
  "align",
]
`;

exports[`Header Navigation styled components NavigationList NavigationList: NavigationList has correct styles when set to align flex-start 1`] = `
Array [
  "align",
]
`;

exports[`Header Navigation styled components StyledRoot Root: StyledRoot has correct styles 1`] = `
Object {
  "borderBottom": "1px solid $theme.colors.border",
  "display": "flex",
  "fontFamily": "$theme.typography.font400.fontFamily",
  "fontSize": "$theme.typography.font400.fontSize",
  "fontWeight": "$theme.typography.font400.fontWeight",
  "lineHeight": "$theme.typography.font400.lineHeight",
  "paddingBottom": "$theme.sizing.scale500",
  "paddingTop": "$theme.sizing.scale500",
}
`;
